public/js/app.js

```diff
@@ -1,106 +1,82 @@
-                  // but not the actual time.  See https://github.com/hilios/jQuery.countdown/issues/144#issuecomment-146582246)
-                  $("#timer").countdown('pause');
-                  // Clear timer set for next countdown.
-                  clearTimeout(Timer.timeTillNext);
-                  // Calculate the time remaining for current countdown
-                  var timePaused = new Date().getTime();
-                  Timer.timeRemaining = Timer.timeRemaining - (timePaused - Timer.timeStarted);
-                  Timer.paused = true;
-              // Unpause countdown if it's currently paused
-              } else {
-                  // Reset current countdown based on time remaining
-                  Timer._countdown(Timer.timeRemaining);
-                  // Reset timer for the next countdown
-                  Timer.timeTillNext = setTimeout( function() {
-                      Timer._setCountdown();
-                  }, Timer.timeRemaining);
-                  Timer.paused = false;
-              }
-          }
-        });
-    },
-    _incremLength: function (locId, direction = 'up') {
-        var currLen = Number($('#' + locId).text());
-        var d = 1;
-        if (direction === 'down')
-            d = -1;
-        currLen = Math.max(0, currLen + d);
-        $('#' + locId).html(currLen);
-    },
-    // Set countdown.  Parameter 'length' is in ms
-    _countdown: function (timeRemaining) {
-        Timer.timeStarted = new Date().getTime();
-        var endTime = new Date().setTime(Timer.timeStarted + timeRemaining);
-        $("#timer").countdown(endTime, function(event) {
-            $(this).text(event.strftime('%H:%M:%S'));
-        });
-        // Only play bell if the Timer._countdown function was called due to a swap between
-        // session and break, not if it was called to unpause.
-        if(Timer.paused === false)
-            Timer.bell.play();
-        Timer.paused = false;
-    },
-    _setCountdown: function(lenLocId, breakLabel) {
-        // Set variables
-        var label = $('#circleLabel').text() ;
-        var lenLocId = (label === 'Start' || label === 'Break') ? '#session-len': '#break-len';
-        var breakLabel = (label === 'Start' || label === 'Break') ? 'Session': 'Break';
-        // Set up countdown
-        Timer.timeRemaining = Number($(lenLocId).text()) * Timer.minute;
-        $('#circleLabel').html(breakLabel);
-        Timer._countdown(Timer.timeRemaining);
-        // Set timer for the following countdown
-        Timer.timeTillNext = setTimeout( function() {
-            Timer._setCountdown();
-        }, Timer.timeRemaining);
-    },
-    _reset: function() {
-        $('#circleLabel').html('Start');
-        clearTimeout(Timer.timeTillNext);
-        if(Timer.paused === false)
-            $('#timer').countdown('stop');
-        $('#timer').html('00:00:00');
-        Timer.paused = true;
```

Comment:
Btw, angular allows you to scale many similar items with ease.  Say I want 3 different clocks a the same time that function independently of each other?  With small changes, you can include a button to add another clock.  Use ng-repeat for this.  Basically restructure your app so that your main app controller keeps track of $scope.clocks = [{<length and labels>}, {<length and labels>}...].  Move the logic in here to another controller file called maybe ClockCtrl.  Then you should be able to leverage ng-repeat in the html to repeat the number of clocks.  You can test by manually changing $scope.clocks.  You can even add a delete button for each clock too.  Hint: each ctrl has its own scope but they all have access to the parent scope.  Use firebug to console.log $scope.